### PR TITLE
Feature/translations

### PR DESCRIPTION
**Pull request by Baklap4**
Added translations (German, Dutch, Japanese, Spanish) to be used within the webserver frontend. see #236 

Closes #263

### CODE REVIEW

**Comment by Baklap4** [comment]
@Lord-Kamina correct i forgot to cherrypick one older commit of yours. It's fixed as of https://github.com/performous/performous/pull/264/commits/f9f7e7b49a9a70b656a9c7acb3f094f0f1595662

File: game/fs.cc [hunk 1]
```diff
@@ -196,12 +196,23 @@ Paths getThemePaths() {
...
 	Paths paths = getPaths();
-	Paths infixes = { themes / theme,
-					  themes / def,
+	Paths infixes = { 
+					  themes / theme,
 					  themes / theme / www,
-					  themes / def / www,
+					  themes / theme / www / js,
+					  themes / theme / www / css,
+					  themes / theme / www / images,
```
Comment (nieknooijens) [inline game/fs.cc]: is this really necessary? or should we program it that it just scans that folder recursively?

File: game/i18n.hh [hunk 2]
```diff
@@ -1,42 +1,27 @@
 #pragma once
 
-#ifdef USE_GETTEXT
-/* Internationalization Dependances */
-    #ifdef __APPLE__ // Using system gettext can choke portability of an app-bundle.
```
Comment (nieknooijens) [inline game/i18n.hh]: glad we got rid of some macros, macros are evil!

File: game/i18n.hh [hunk 3]
```diff
@@ -1,42 +1,27 @@
...
-	static bool enabled() {
-		#ifdef USE_GETTEXT
-		return true;
-		#else
-		return false;
-		#endif
-	}
-};
+	public:
+		Gettext(const char *package) {
```
Comment (nieknooijens) [inline game/i18n.hh]: we got rid of gettext right? please right-click-> refactor->rename symbol

File: game/webserver.cc [hunk 4]
```diff
@@ -96,7 +99,23 @@ http_server::response WebServer::GETresponse(const http_server::request &request
...
 		return http_server::response::stock_reply(http_server::response::ok, std::to_string(config["game/playlist_screen_timeout"].i()));
-	} else {
+	} else if(request.destination.find("/api/language") == 0) {
+		map<std::string, std::string> localeMap = GenerateLocaleDict();
+		
+		Json::Value jsonRoot = Json::objectValue;
+		for (auto const &kv : localeMap) {
+			std::string key = kv.first;
+			//Hack to get an easy key value pair within the json object.
+			if(key == "Web interface by Niek Nooijens and Arjan Speiard, for full credits regarding Performous see /docs/Authors.txt"){
```
Comment (nieknooijens) [inline game/webserver.cc]: suggestion: hyperlink where it just links to authors.txt?

Comment (Baklap4) [inline game/webserver.cc]: markup can be done within the po files for this specific one imho this specific string is placed within a div so the link will be visible.

File: game/fs.cc (hunk 1 above)
Comment (Baklap4) [inline game/fs.cc]: I've did this in preperation of #251 
I think the themes folder should be specified, and each folder (and thus each theme) should be scanned

File: game/webserver.cc (hunk 4 above)
Comment (Baklap4) [inline game/webserver.cc]: @nieknooijens One problem here, docs/authors.txt does not live within the scope of the webserver, therefore is not reachable or we should symlink it to the www folder, i'm not quite fond of that..

**Pull request merged**